src/main/java/com/example/provider/json/VehicleSerializer.java

```diff
@@ -21,43 +21,66 @@
 import com.google.gson.JsonElement;
 import com.google.gson.JsonSerializationContext;
 import com.google.gson.JsonSerializer;
+import com.google.protobuf.Int32Value;
+
 import google.maps.fleetengine.v1.Vehicle;
 import java.lang.reflect.Type;
 import java.util.List;
 
 /**
  * Serializer for vehicle object to provide relevant information to its clients.
  *
- * <p>From the fleetengine vehicle object, this serializer retrieves the name and the vehicleState.
+ * <p>
+ * From the fleetengine vehicle object, this serializer retrieves the name and
+ * the vehicleState.
  */
 final class VehicleSerializer implements JsonSerializer<Vehicle> {
 
+  Int32Value kDefaultInt32Value = Int32Value.newBuilder()
+      .setValue(0)
+      .build();
+
   @Override
   public JsonElement serialize(Vehicle src, Type typeOfSrc, JsonSerializationContext context) {
-    List<Waypoint> waypoints =
-        src.getWaypointsList().stream()
-            .map(
-                waypoint ->
-                    Waypoint.newBuilder()
-                        .setTripId(waypoint.getTripId())
-                        .setWaypointType(
-                            WaypointUtils.getWaypointTypeString(waypoint.getWaypointType()))
-                        .setLocation(
-                            SerializationUtils.createSerializedLocation(
-                                waypoint.getLocation().getPoint()))
-                        .build())
-            .collect(toList());
+    List<Waypoint> waypoints = src.getWaypointsList().stream()
+        .map(
+            waypoint -> Waypoint.newBuilder()
+                .setTripId(waypoint.getTripId())
+                .setWaypointType(
+                    WaypointUtils.getWaypointTypeString(waypoint.getWaypointType()))
+                .setLocation(
+                    SerializationUtils.createSerializedLocation(
+                        waypoint.getLocation().getPoint()))
+                .build())
+        .collect(toList());
+
+    SerializedLatLng serializedLatLng = SerializedLatLng.newBuilder()
+        .setLatitude(src.getLastLocation().getLocation().getLatitude())
+        .setLongitude(src.getLastLocation().getLocation().getLongitude())
+        .build();
+
+    SerializedLastLocation.Builder serializedLastLocationBuilder = SerializedLastLocation.newBuilder()
+        .setLatLng(serializedLatLng);
+
+    if (src.getLastLocation().hasHeading()) {
```

Comment:
This can be reduced to: 

```
int vehicleHeading = src.getLastLocation.hasHeading() ? src.getLastLocation.getHeading().getValue() : 0;
serializedLastLocationBuilder.setHeading(vehicleHeading);
```